### PR TITLE
Fixed #28109 -- Adjusted stacklevel of warnings raised when paginating unordered querysets.

### PR DESCRIPTION
**Pull request by charettes**
Refs #26290.

### CODE REVIEW

**Comment by charettes** [comment]
@timgraham makes sense I'll adjust the release notes. I gave a try at testing it as well but abandoned after 20 minutes spent on it. Having slept on it I think the best avenue would be prevent the warning from being converted to an exception for the duration of the test but I haven't tried it yet. I'll give it a shot later today and if I don't come up with a solution I'll just commit it without a test 🤷‍♂️ .

**Comment by timgraham** [comment]
I couldn't figure out why `assertRaisesMessage()` works there (why is the warning converted to an exception). Or is it... because the test passes with `assertWarns()` also.

**Comment by timgraham** [comment]
Oh, `warnings.simplefilter("error", RuntimeWarning)` in `runtests.py`.

**Comment by charettes** [comment]
yeah it's caused by the `warnings.simplefilter('error', RuntimeWarning)` call. I was thinking of doing a `warnings.simplefilter('once', UnorderedObjectListWarning)` in a `warnings.catch_warnings()` context and play around with `module`, `file` and `lineno` to assert it's working correctly.

File: tests/pagination/tests.py [hunk 1]
```diff
@@ -318,12 +319,16 @@ def test_page_getitem(self):
...
             "Pagination may yield inconsistent results with an unordered "
             "object_list: <QuerySet [<Article: Article 1>, "
             "<Article: Article 2>, <Article: Article 3>, <Article: Article 4>, "
             "<Article: Article 5>, <Article: Article 6>, <Article: Article 7>, "
             "<Article: Article 8>, <Article: Article 9>]>"
-        )
-        with self.assertRaisesMessage(UnorderedObjectListWarning, msg):
-            Paginator(Article.objects.all(), 5)
+        ))
+        self.assertEqual(warning.filename, __file__)
```
Comment (charettes) [inline tests/pagination/tests.py]: @timgraham that's the assertion that makes sure the warning actually points at the `Paginator` caller.

File: tests/pagination/tests.py [hunk 2]
```diff
@@ -318,12 +319,16 @@ def test_page_getitem(self):
         self.assertIsInstance(p.object_list, list)
 
     def test_paginating_unordered_queryset_raises_warning(self):
-        msg = (
+        with warnings.catch_warnings(record=True) as warns:
+            warnings.filterwarnings('always', category=UnorderedObjectListWarning)
```
Comment (timgraham) [inline tests/pagination/tests.py]: `# Prevent the RuntimeWarning subclass from appearing as an exception due to the warnings.simplefilter() in runtests.py.`

File: tests/pagination/tests.py (hunk 1 above)
Comment (timgraham) [inline tests/pagination/tests.py]: Looks good but if you think you have to explain it, maybe a comment to the effect of ``# The warning points at the Paginator caller (i.e. the stacklevel is appropriate).`` would be appropriate.

**Pull request merged**